cms/bundles/wagtail_hooks.py

```diff
@@ -0,0 +1,199 @@
+from functools import cached_property
+from typing import TYPE_CHECKING, Any, Union
+
+from django.db.models import QuerySet
+from django.urls import include, path
+from django.utils.timezone import now
+from django.utils.translation import gettext_lazy as _
+from wagtail import hooks
+from wagtail.admin.ui.components import Component
+from wagtail.admin.widgets import PageListingButton
+from wagtail.log_actions import LogFormatter
+from wagtail.permission_policies import ModelPermissionPolicy
+
+from . import admin_urls
+from .models import Bundle, BundledPageMixin
+from .viewsets import bundle_chooser_viewset, bundle_viewset
+
+if TYPE_CHECKING:
+    from django.http import HttpRequest
+    from django.urls import URLPattern
+    from django.urls.resolvers import URLResolver
+    from laces.typing import RenderContext
+    from wagtail.log_actions import LogActionRegistry
+    from wagtail.models import ModelLogEntry, Page
+
+    from cms.users.models import User
+
+
+@hooks.register("register_admin_viewset")
+def register_viewset() -> list:
+    """Registers the bundle viewsets.
+
+    @see https://docs.wagtail.org/en/stable/reference/hooks.html#register-admin-viewset
+    """
+    return [bundle_viewset, bundle_chooser_viewset]
+
+
+class PageAddToBundleButton(PageListingButton):
+    """Defines the 'Add to Bundle' button to use in different contexts in the admin."""
+
+    label = _("Add to Bundle")
+    icon_name = "boxes-stacked"
+    aria_label_format = _("Add '%(title)s' to a bundle")
+    url_name = "bundles:add_to_bundle"
+
+    @property
+    def permission_policy(self) -> ModelPermissionPolicy:
+        """Informs the permission policy to use Bundle-derived model permissions."""
+        return ModelPermissionPolicy(Bundle)
+
+    @property
+    def show(self) -> bool:
+        """Determines whether the button should be shown.
+
+        We only want it for pages inheriting from BundledPageMixin that are not in an active bundle.
+        """
+        if not isinstance(self.page, BundledPageMixin):
+            return False
+
+        if self.page.in_active_bundle:
+            return False
+
+        # Note: limit to pages that are not in an active bundle
+        can_show: bool = (
+            self.page_perms.can_edit() or self.page_perms.can_publish()
+        ) and self.permission_policy.user_has_any_permission(self.user, ["add", "change", "delete"])
+        return can_show
```

Comment:
We need to update the logic to exclude pages that are "Live."

Reply:
This is still outstanding.

Reply:
The comment from https://github.com/ONSdigital/dis-wagtail/pull/33#discussion_r1863392065 ('it is more nuanced') also applies. and is on CMS-232

The gist of it is, and will confirm with Paula: should we exclude all live pages (if so, what about corrections)? Or only those that are published and has no pending drafts?